src/Solucion_JavierRamiro/src/programacion/empresaprogramacion/Empresa.java

```diff
@@ -0,0 +1,132 @@
+            System.out.println(empleado);
+        }
+    }
+
+    public void toStringOrdenNombre() {
+        Collections.sort(empleados, new Comparator<Empleado>() {
+            @Override
+            public int compare(Empleado emp1, Empleado emp2) {
+                return emp1.getNombre().compareTo(emp2.getNombre());
+            }
+        });
+
+        for (Empleado empleado : empleados) {
+            System.out.println(empleado);
+        }
+    }
+
+    public void cargarDesdeCSV(String archivo) {
+        try (BufferedReader br = new BufferedReader(new FileReader(archivo))) {
+            String linea;
+            while ((linea = br.readLine()) != null) {
+                String[] partes = linea.split(",");
+                if (partes.length != 5) {
+                    throw new ParametroInvalidoException("Número incorrecto de parámetros en la línea: " + linea);
+                }
+
+                String tipoStr = partes[0];
+                String nombre = partes[1];
+                double sueldo = Double.parseDouble(partes[2]);
+                String lenguaje = partes[3];
+                Boolean.parseBoolean(partes[4]);
+
+                TipoEmpleado tipo = TipoEmpleado.valueOf(tipoStr.toUpperCase());
+
+                Empleado empleado;
+                if (tipo == TipoEmpleado.GERENTE) {
+                    String departamento = lenguaje;
+                    empleado = new Gerente(nombre, sueldo, departamento);
+                } else if (tipo == TipoEmpleado.PROGRAMADOR) {
+                    empleado = new Programador(nombre);
+                } else {
+                    throw new ParametroInvalidoException("Tipo de empleado inválido en la línea: " + linea);
+                }
+
+                empleados.add(empleado);
+            }
+            System.out.println("Datos cargados desde el archivo CSV correctamente.");
+        } catch (IOException e) {
+            System.out.println("Error al leer el archivo CSV: " + e.getMessage());
+        } catch (NumberFormatException e) {
+            System.out.println("Error al convertir el sueldo a número en el archivo CSV: " + e.getMessage());
+        } catch (ParametroInvalidoException e) {
+            System.out.println("Error en los parámetros del archivo CSV: " + e.getMessage());
+        } catch (IllegalArgumentException e) {
+            System.out.println("Error en el tipo de empleado en el archivo CSV: " + e.getMessage());
+        }
+    }
+
+    public void guardarEnCSV(String archivo) {
+        try (BufferedWriter bw = new BufferedWriter(new FileWriter(archivo))) {
+            for (Empleado empleado : empleados) {
+                String tipoStr = empleado.getTipo().name().toLowerCase();
+                String nombre = empleado.getNombre();
+                double sueldo = empleado.getSueldo();
+                String lenguaje = "";
+                String plusIdioma = "";
+
+                if (empleado.getTipo() == TipoEmpleado.GERENTE) {
```

Comment:
Ojo, quizá en este caso coincida el número de columnas, pero no es lo mismo el atributo lenguaje de un Programador que el atributo departamento de un Gerente. Lo suyo sería que para los programadores se usase un fichero csv y para los gestores otro fichero csv.

Creo que la analogía sería, según lo que creo que habéis visto en BBDD, si los guardaríais en la misma tabla o en tablas separadas.